test/unit/storage_test.rb

```diff
@@ -114,14 +102,73 @@ def test_sentinels_malformed_url
     end
   end
 
+  def test_sentinels_simple_url
+    config_obj = {
+      url: 'master-name', # url of the sentinel master name conf
+      sentinels: 'redis://127.0.0.1:26379'
+    }
+    conn = Storage.send :orig_new, Storage::Helpers.config_with(config_obj)
+    assert_sentinel_connector(conn.client)
+    assert_client_config(conn.client, url: "redis://#{config_obj[:url]}",
+                                      sentinels: [{ host: '127.0.0.1', port: 26_379 }])
+  end
+
+  def test_sentinels_correct_role
+    %i[master slave].each do |role|
+      config_obj = {
+        url: 'redis://master-group',
+        sentinels: 'redis://127.0.0.1:26379',
+        role: role
+      }
+      conn = Storage.send :orig_new, Storage::Helpers.config_with(config_obj)
+      assert_sentinel_connector(conn.client)
+      assert_client_config(conn.client, url: config_obj[:url],
+                                        sentinels: [{ host: '127.0.0.1', port: 26_379 }],
+                                        role: role)
+    end
+  end
+
+  def test_sentinels_role_empty
+    [''.to_sym, '', nil].each do |role|
+      config_obj = {
+        url: 'redis://127.0.0.1:6379/0',
+        sentinels: 'redis://127.0.0.1:26379',
```

Comment:
perhaps also add a test where `sentinels` is `"    "` and/or `",,,," and combined with `role` having a valid value, like the comment I do elsewhere wondering about what would happen? (ie. this client breaks in the weird case of specifying sentinels as empty, so I fear this could also be problematic).

Reply:
no problem with that. When ```sentinels``` is ```" "``` or ```,,,```, then redis connection will be called with empty array.
```
con = Redis.new(url: 'redis://mymaster', sentinels: [], role: :master)
```
No matter what is ```role``` value, error will be raised reporting error about sentinels configuration:
```
con = Redis.new(url: 'redis://mymaster', sentinels: [], role: :master)
=> #<Redis client v3.3.3 for redis://mymaster:6379/0>
irb(main):003:0> con.get('3')
Redis::CannotConnectError: No sentinels available.
```

If no sentinels is configured, ```role``` will not be read and will work as usual.

Reply:
ok :+1: